q2cli/builtin/tools.py

```diff
@@ -604,3 +604,247 @@ def citations(path):
+              help='Path to an existing cache to check the status of.')
+def cache_status(path):
+    from qiime2.core.cache import Cache
+    from qiime2.sdk.result import Result
+
+    from q2cli.core.config import CONFIG
+
+    data_output = []
+    pool_output = []
+    try:
+        cache = Cache(path)
+        with cache.lock:
+            for key in cache.get_keys():
+                key_values = cache.read_key(key)
+
+                if (data := key_values['data']) is not None:
+                    data_output.append(
+                        'data: %s -> %s' %
+                        (key, str(Result.peek(cache.data / data))))
+                elif (pool := key_values['pool']) is not None:
+                    pool_output.append(
+                        'pool: %s -> size = %s' %
+                        (key, str(len(os.listdir(cache.pools / pool)))))
+    except Exception as e:
+        header = "There was a problem getting the status of the cache at " \
+                 "path '%s':" % path
+        q2cli.util.exit_with_error(e, header=header, traceback=None)
+
+    if not data_output:
+        data_output = 'No data keys in cache'
+    else:
+        data_output = '\n'.join(data_output)
+        data_output = 'Data keys in cache:\n' + data_output
+
+    if not pool_output:
+        pool_output = 'No pool keys in cache'
+    else:
+        pool_output = '\n'.join(pool_output)
+        pool_output = 'Pool keys in cache:\n' + pool_output
+
+    output = data_output + '\n\n' + pool_output
+    success = "Status of the cache at the path '%s':\n\n%s" % (path, output)
+    click.echo(CONFIG.cfg_style('success', success))
+
+
+@tools.command(name='cache-validate',
+               short_help='Validates the contents of the cache.',
+               help='Fully validates all data in the cache using each '
+                    'artifacts validate functions and checksums.',
+               cls=ToolCommand)
+@click.option('--path', required=True,
+              type=click.Path(exists=True, file_okay=False, dir_okay=True,
+                              readable=True),
+              help='Path to an existing cache to validate.')
+def cache_validate(path):
+    from qiime2.core.cache import Cache
+    from qiime2.core.util import is_uuid4
+    from qiime2 import Artifact
+
+    from q2cli.core.config import CONFIG
+
+    click.echo(
+        CONFIG.cfg_style('success', "Validating cache at '%s'\n" % path))
+
+    try:
+        cache = Cache(path)
+        with cache.lock:
+            for data in cache.get_data():
+                try:
+                    click.echo(
+                        CONFIG.cfg_style('success', 'Validating: %s' % data))
+                    if not is_uuid4(data):
+                        raise ValueError(
+                            "Item in data directory '%s' is not a valid "
+                            "uuid4." % data)
+
+                    art = Artifact.load(cache.data / data)
```

Comment:
I thought we had made it so that this wasn't possible without going through the `Cache` object.

I would have expected something like `a = Cache.load(...); a.validate()`.

Perhaps we should have a `Cache.validate()` method instead?